src/services/addressResolver/index.ts

```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+import { ThirdPartyAPIEntryProps } from "../../common/hooks/useThirdPartyAPIEndpoints";
+
+export const resolveAddressNameByEndpoint = async (url: string) => {
+  try {
+    const response = await axios.get(url);
+    return response.data.identity.name;
```

Comment:
```suggestion
    return response.data?.identity?.name;
```

you cant be certain that the response is correct